echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java

```diff
@@ -57,4 +60,27 @@
     private String pubsubSystem;
     private String subscriptionName;
   }
+
+  @RequestMapping(value = "/publishers", method = RequestMethod.GET)
+  List<PubsubPublishersBySystem> getPublishers() {
+    return pubsubPublishers
+      .getAll()
+      .stream()
+      .map(p -> PubsubPublishersBySystem.builder()
+        .pubsubSystem(p.getPubsubSystem().toString())
+        .publisherName(p.getName())
+        .topicName(p.getTopicName())
+        .build())
+      .collect(Collectors.toList());
+  }
+
+  @Data
+  @Builder
+  @NoArgsConstructor
```

Comment:
Minor point; I think that `@Builder` implies `@AllArgsConstructor` (which is what it uses) so we could remove that and also remove the `@NoArgsConstructor`, though I think in practice we tend to add all these annotations to data classes.

Reply:
Removed both.